x-pack/legacy/plugins/code/server/distributed/multinode/non_code_node_adapter.ts

```diff
@@ -100,16 +104,16 @@ export class NonCodeNodeAdapter implements ServiceHandlerAdapter {
         });
       } catch (e) {
         this.log.error('parse json failed: ' + buffer.toString());
-        throw Boom.boomify(e, { statusCode: 500 });
+        throw e;
       }
     } else {
       this.log.error(
         `received ${res.statusCode} from ${baseUrl}/${path}, params was ${util.inspect(
           payload.params
         )}`
       );
-      const body: Boom.Payload = await Wreck.read(res, { json: true });
-      throw new Boom(body.message, { statusCode: res.statusCode || 500, data: body.error });
+      const body = await Wreck.read(res, { json: true });
+      throw new Error(body.message);
```

Comment:
I'm not sure removing `statusCode` is right here, some status code is important, such as 404

Reply:
I agree simply switch from Boom to Error will change some status code, but I anyway do it for the following reasons:

1. we nested the Boom HTTP status code deeply into our internal service is not a good idea. This will make the internal service coupled with the http server layer very tightly. This is will make some future changes (like this PR) quite difficult to proceed.
2. Ideally, we should only append status code in the http server layer, i.e. the route definitions, while define our own Error types (e.g. `LangServerNotInitializedError()` so that the route definition could assemble the right status code appropriately.
3. I think our frontend does not rely on the status code too much that a 404 handling will be the same as 500 (@spacedragon need your confirmation). So I anyway make this change. 

Thoughts?

Reply:
At least,  404 is used for that file not found page.

Reply:
@spacedragon revert back the changes of removing Boom. let's figure this out later. Also, skippe the 2 unit tests here: https://github.com/elastic/kibana/blob/5015a368c3fd581e115248cf85b32941c0f87047/x-pack/legacy/plugins/code/server/distributed/code_services.test.ts#L65. Let's chat offline and fix it later.